src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -47,9 +47,18 @@ private sealed class Parser
 #pragma warning restore
 
             public List<DiagnosticInfo> Diagnostics { get; } = new();
+            private Location? _contextClassLocation;
 
             public void ReportDiagnostic(DiagnosticDescriptor descriptor, Location? location, params object?[]? messageArgs)
             {
+                Debug.Assert(_contextClassLocation != null);
+
+                if (location is null || !_knownSymbols.Compilation.ContainsLocation(location))
+                {
+                    // If location is null or is a location outside of the current compilation, fall back to the location of the context class.
+                    location = _contextClassLocation;
```

Comment:
>_contextClassLocation [](http://example.com/codeflow?start=31&length=21)

could you please assert here this is not null?

Reply:
There's one at the top of this method, line 54.